src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
@lorenzo I forgot about [Oracle Driver](https://github.com/CakeDC/cakephp-oracle-driver). While oracle supports EXPLAIN statement, but as far as I know, it would requires two statements:
```sql
EXPLAIN PLAN FOR SELECT * FROM "users";
SELECT * FROM PLAN_TABLE;
```
Calling `explainSQL()` wouldn't be suitable for Oracle. Do you have any thoughts?